platform/measurement_utils.cpp

```diff
@@ -18,15 +18,21 @@
 
 namespace measurement_utils
 {
-std::string ToStringPrecision(double d, int pr)
+// Global Locale variable to store system locale.
+static platform::Locale g_systemLocale = {
+        "",  // Undefined language.
+        "",  // Undefined country.
+        "",  // Undefined currency.
+        ".", // Dot as default decimal separator.
+        platform::kNarrowNonBreakingSpace // Narrow space as default grouping (thousands) separator.
+};
+
+void RefreshSystemLocale()
 {
-  // We assume that the app will be restarted if a user changes device's locale.
-  static auto const locale = platform::GetCurrentLocale();
-
-  return ToStringPrecisionLocale(locale, d, pr);
+  g_systemLocale = platform::GetCurrentLocale();
 }
 
-std::string ToStringPrecisionLocale(platform::Locale const & loc, double d, int pr)
```

Comment:
The previous implementation of a function allowed to unit-test it nicely. Would it be better to pass the g_systemLocale as a parameter?

Reply:
> The previous implementation of a function allowed to unit-test it nicely. Would it be better to pass the g_systemLocale as a parameter?

I've returned to the previous implementation, leaving the function `ToStringPrecisionLocale()` and also I've added unit tests for the function `ToStringPrecision()`.

I've had to set the default grouping separator as `comma` in order for the `distance_tests` to work.